docs/_tooling/extensions/score_metamodel/checks/check_options.py

```diff
@@ -34,29 +33,29 @@ def get_need_type(needs_types: list[NeedsInfoType], directive: str):
 # req-traceability: TOOL_REQ__toolchain_sphinx_needs_build__options
 @local_check
 def check_options(
+    app: Sphinx,
     need: NeedsInfoType,
     log: CheckLogger,
-    needs_types=production_needs_types,
 ):
     """
     Checking if all described and wanted attributes are present and their values follow the described pattern.
     """
-
+    needs_types = app.config.needs_types
     try:
         need_options = get_need_type(needs_types, need["type"])
     except ValueError:
         msg = f"with type `{need['type']}`: no type info defined for semantic check."
         log.warning_for_option(need, log, msg)
         return
 
-    required_options: list[tuple[str, str]] = need_options.get("req_opt", [])
-    optional_options: list[tuple[str, str]] = need_options.get("opt_opt", [])
+    required_options: dict[str, str] = need_options.get("mandatory_options", {})
+    optional_options: dict[str, str] = need_options.get("opt_opt", {})
 
     if len(required_options) == 0:
         msg = "no type info defined for semantic check."
-        log.warning_for_option((need, "type", msg))
+        log.warning_for_option(need, "type", msg)
```

Comment:
Was there no mypy/pyright warning here?

Reply:
Nah it's fine. We'll get warnings in the future. It's just a reminder to hurry up with linters.